Sources/Purchasing/OfferingsManager.swift

```diff
@@ -38,7 +38,7 @@ class OfferingsManager {
     }
 
     func offerings(appUserID: String, completion: ((Result<Offerings, Error>) -> Void)?) {
-        guard let cachedOfferings = deviceCache.cachedOfferings else {
+        guard let cachedOfferings = self.deviceCache.cachedOfferings else {
```

Comment:
[not needed for this PR]
Here's the other bug that I was talking about in slack: we assume that not having cached offerings means that we need to request them, but a request might already be in flight. 
So here the check should have been something like 
`if cache is not stale then append to callbacks`, to account for a possible request in flight (this is why we set the cache timestamp to now before we actually have the cache returned from the backend). 

Then again this concurrency control predates the request caching that we have now, which automatically takes care of this bit. It'd be nice to move the bits that do `setOfferingsCacheTimestampToNow` to only do that after we actually have some data from the backend, but that's not needed for now. 

Same thinking for CustomerInfo, they both had the same system in place

Reply:
I sent you a message about this, because I _think_ this solves that as well.
Merging this PR for now but let me know if I misunderstood and happy to write a failing test for that scenario as well.

Reply:
See #1839.